### PR TITLE
Replace recursion flag from boolean to int

### PR DESCRIPTION
**Pull request by badosu**
Max depth controlled by MAX_CMD_RECURSION_DEPTH

### CODE REVIEW

File: rts/Lua/LuaSyncedCtrl.h [hunk 1]
```diff
@@ -5,21 +5,23 @@
...
 	private:
-		static bool inCreateUnit;
-		static bool inDestroyUnit;
-		static bool inTransferUnit;
-		static bool inCreateFeature;
-		static bool inDestroyFeature;
-		static bool inGiveOrder;
-		static bool inHeightMap;
-		static bool inSmoothMesh;
+		static int inCreateUnit;
```
Comment (lhog) [inline rts/Lua/LuaSyncedCtrl.h:17]: While you are on it please convert from `static int` into `inline static int` it's more modern and compact.
Eg. `inline static int inDestroyUnit = 0;`

File: rts/Lua/LuaSyncedCtrl.cpp [hunk 2]
```diff
@@ -1243,7 +1243,7 @@ int LuaSyncedCtrl::CreateUnit(lua_State* L)
 {
 	CheckAllowGameChanges(L);
 
-	if (inCreateUnit) {
+	if (inCreateUnit >= MAX_CMD_RECURSION_DEPTH) {
 		luaL_error(L, "[%s()]: recursion is not permitted", __func__);
```
Comment (lhog) [inline rts/Lua/LuaSyncedCtrl.cpp:1247]: Perhaps add the maximum recursion depth to the diagnostic value here and below?

File: rts/Lua/LuaSyncedCtrl.cpp [hunk 3]
```diff
@@ -84,14 +84,14 @@ using std::min;
...
 
-bool LuaSyncedCtrl::inCreateUnit = false;
-bool LuaSyncedCtrl::inDestroyUnit = false;
-bool LuaSyncedCtrl::inTransferUnit = false;
-bool LuaSyncedCtrl::inCreateFeature = false;
-bool LuaSyncedCtrl::inDestroyFeature = false;
-bool LuaSyncedCtrl::inGiveOrder = false;
-bool LuaSyncedCtrl::inHeightMap = false;
-bool LuaSyncedCtrl::inSmoothMesh = false;
+int LuaSyncedCtrl::inCreateUnit = 0;
```
Comment (lhog) [inline rts/Lua/LuaSyncedCtrl.cpp:87]: See `inline static` ... comment below

File: rts/Lua/LuaSyncedCtrl.cpp [hunk 4]
```diff
@@ -84,14 +84,14 @@ using std::min;
...
+int LuaSyncedCtrl::inDestroyUnit = 0;
+int LuaSyncedCtrl::inTransferUnit = 0;
+int LuaSyncedCtrl::inCreateFeature = 0;
+int LuaSyncedCtrl::inDestroyFeature = 0;
+int LuaSyncedCtrl::inGiveOrder = 0;
+int LuaSyncedCtrl::inHeightMap = 0;
+int LuaSyncedCtrl::inSmoothMesh = 0;
```
[the remark this answers is not in the document]
Comment (badosu) [reply rts/Lua/LuaSyncedCtrl.cpp:94]: I'm not, glad to revert to `bool`

[the remark this answers is not in the document]
Comment (badosu) [reply rts/Lua/LuaSyncedCtrl.cpp:94]: Redefining as either `inline static` or `inline static int`, together with the `.h` change to `inline static int` throws:

```
/home/badosu/Code/spring/rts/Lua/LuaSyncedCtrl.cpp:87:8: error: ‘static’ may not be used when defining (as opposed to declaring) a static data member [-fpermissive]
   87 | inline static int LuaSyncedCtrl::inCreateUnit = 0;
      |        ^~~~~~
/home/badosu/Code/spring/rts/Lua/LuaSyncedCtrl.cpp:87:19: error: redefinition of ‘int LuaSyncedCtrl::inCreateUnit’
   87 | inline static int LuaSyncedCtrl::inCreateUnit = 0;
      |                   ^~~~~~~~~~~~~
In file included from /home/badosu/Code/spring/rts/Lua/LuaSyncedCtrl.cpp:6:
/home/badosu/Code/spring/rts/Lua/LuaSyncedCtrl.h:17:35: note: ‘int LuaSyncedCtrl::inCreateUnit’ previously declared here
   17 |                 inline static int inCreateUnit;
      |
```

Suggestions?

**Comment by badosu** [comment]
@lhog Let me know if there are any pending issues

File: rts/Lua/LuaSyncedCtrl.cpp [hunk 5]
```diff
@@ -84,12 +84,7 @@ using std::min;
 /******************************************************************************/
 /******************************************************************************/
 
-bool LuaSyncedCtrl::inCreateUnit = false;
-bool LuaSyncedCtrl::inDestroyUnit = false;
 bool LuaSyncedCtrl::inTransferUnit = false;
```
Comment (lhog) [inline rts/Lua/LuaSyncedCtrl.cpp:87]: Move these into class as well please (as `inline static...`)

**Pull request merged**